src/python/pants/backend/python/target_types.py

```diff
@@ -167,6 +167,10 @@ class PythonRunGoalUseSandboxField(TriBoolField):
         If false, runs of this target with the `run` goal will use the in-repo sources
         directly.
 
+        Note that this field is going to apply only when running a target with the `run` goal.
+        Invoking the `test` goal on a `python_test` target with the field set to false won't
+        change the fact that tests are always executed in a sandbox.
```

Comment:
```suggestion
        Note that this field is only applies when running a target with the `run` goal.
        No other goals (such as `test`, if applicable) consult this field.
```

Reply:
Thanks, Josh. I've adopted your wording, this is very concise and conveys the message I had in mind.